app/main.py

```diff
@@ -1,3 +1,58 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as config:
+        data = json.load(config)
+
+    fuel_price = data["FUEL_PRICE"]
+    customers_data = data["customers"]
+    shops_data = data["shops"]
+
+    for customer_data in customers_data:
+        customer = Customer(
+            customer_data["name"],
+            customer_data["product_cart"],
+            customer_data["location"],
+            customer_data["money"],
+            customer_data["car"]
+        )
+
+        print(f"{customer.name} has {customer.money} dollars")
+
+        relevant_shop = None
+        relevant_shop_trip_costs = float("inf")
+        for shop_data in shops_data:
+            shop = Shop(
+                shop_data["name"],
+                shop_data["location"],
+                shop_data["products"]
+            )
```

Comment:
```suggestion
            shop = Shop(**shop_data)
```
you can use unpacking here.